src/streams.h

```diff
@@ -138,6 +138,80 @@ class CVectorWriter
     size_t nPos;
 };
 
+/* Minimal stream for reading from an existing vector by reference
+ */
+class CVectorReader
+{
+private:
+    const int nType;
+    const int nVersion;
+    const std::vector<unsigned char>& vchData;
```

Comment:
In commit "streams: Create CVectorReader stream interface for vectors." (93f702b08e413c5c025b155bfb62b721d27939f5)

It would be nice if this just had `const unsigned char*` and `size_t` members instead of a requiring a reference to an actual vector. That way the class could be used to efficiently deserialize from any memory location, and be compatible with other containers like `std::string`.

Reply:
I'd rather not deal with raw pointers because it leaves space for unsafe accesses. If generality is a concern, I'd prefer a templated approach with random access iterators.

Reply:
Re: https://github.com/bitcoin/bitcoin/pull/12254#discussion_r174888569

I think it would be better to replace VectorReader with SpanReader at this point. It's a simple change that would make this class more flexible and reusable.